diff --git a/CMakeLists.txt b/CMakeLists.txt
index 19038c8..a6e4ee4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 # vim:noexpandtab:
-cmake_minimum_required(VERSION 3.6)
+cmake_minimum_required(VERSION 3.5)
 project(PPSSPP)
 
 #This is supposed to work but doesn't!
@@ -123,7 +123,7 @@ else()
 	set(CoreLinkType STATIC)
 endif()
 
-include_directories(ext/native)
+include_directories(ext/native /opt/vc/lib/libzip/include)
 
 if(NOT OPENGL_LIBRARIES AND USING_GLES2)
 	set(OPENGL_LIBRARIES GLESv2)
